src/main/java/com/Perk.java

```diff
@@ -12,50 +12,29 @@
 public class Perk {
     /* Perk fields */
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    private int id;
+    @Column(unique=true)
     private String name;
```

Comment:
Can we also change the name field to code? It makes more sense that this is the code for the offer instead of the name which would be the same as the description

Reply:
That would make sense, but I think we should separate that as another issue

Reply:
I will create that issue right now